src/components/Nav/SubNav/SubNavItem.js

```diff
@@ -1,11 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router';
 import styled from 'styled-components';
 
-function SubNavItem(props) {
+import { DEALS, MYPAGE } from './navTabList';
+
+function SubNavItem({ children }) {
+  const [selectedTab, setSelectedTab] = useState(null);
+
+  const location = useLocation();
+  const history = useHistory();
+
+  const handleTabClick = (path, id) => {
+    history.push(path);
+    setSelectedTab(id);
+  };
+
+  useEffect(() => {
+    const pathArr = [...DEALS, ...MYPAGE];
+    const index = pathArr.findIndex(el => el.path === location.pathname);
+    const id = pathArr[index].id;
+
+    setSelectedTab(id);
+  }, []);
+
   return (
     <>
-      {props.children.map((li, i) => (
-        <Item key={i}>{li}</Item>
+      {children.map(li => (
+        <Item
+          onClick={() => handleTabClick(li.path, li.id)}
+          key={li.id}
+          id={li.id}
+          selectedTab={selectedTab}
+        >
+          {li.name}
+        </Item>
```

Comment:
선택됬는지를 표현하기 위해서 굳이 selectedTab이라는 state를 만들어서 관리할 필요 없을 것 같습니다!
location.pathname을 이용해서 Item이 선택되어있는지(pathname이 동일한지) 검사하는 것 같은데 저런 로직 필요없어보입니다!
id도 필요없어보입니다, id selector로 적용되고있어서 검사하는 로직에 사용되는 것 이외에도 id selector로 사용되는 side effect도 발생할 수 있어서 지양하는게 좋아보이네요!
```suggestion
        <Item
          onClick={() => handleTabClick(li.path, li.id)}
          key={li.id}
          isSelected={location.pathname === li.path}
        >
          {li.name}
        </Item>
```

Reply:
history 객체 안쓰고 Link 컴포넌트로 바꿀수도 있어보이네요!

Reply:
이 로직을 넣기위해 따로 컴포넌트로 분리한 것 같은데, 그러면 굳이 분리안하고 SubNav.js로 합칠 수도 있어보이네요!